File: single-language-analysis/dataset-files/python/test_smell_043.py
import pytest

class TestVoteTest:
    def test_smell_043(self):
        vote = Vote(3)

        def thread_one():
            try:
                time.sleep(1)
                assert vote.vote("1") == "0"
            except InterruptedException as e:
                raise AssertionError(e)

        def thread_two():
            try:
                time.sleep(0.5)
                assert vote.vote("0") == "0"
            except InterruptedException as e:
                raise AssertionError(e)

        threading.Thread(target=thread_one).start()
        threading.Thread(target=thread_two).start()
        assert vote.vote("0") == "0"

File: single-language-analysis/dataset-files/java/TestSmell043.java
// Original URL: https://github.com/INFO-ESIPE/TP-Programmation_Concurrente/blob/48457a69d6f38d906dda5d47965b1e91bbcb3194/TP%204/src/VoteTest.java#L8-L28

public class TestSmell043 {
    @Test
    public void SimpleVote() throws Exception {
        var vote = new Vote(3);
        Thread.ofPlatform().start(() -> {
            try {
                Thread.sleep(1_000);
                assertEquals("0", vote.vote("1"));
            } catch (InterruptedException e) {
                throw new AssertionError(e);
            }
        });
        Thread.ofPlatform().start(() -> {
            try {
                Thread.sleep(500);
                assertEquals("0", vote.vote("0"));
            } catch (InterruptedException e) {
                throw new AssertionError(e);
            }
        });
        assertEquals("0", vote.vote("0"));
    }
}
